src/util/psbt/map/output.rs

```diff
@@ -55,6 +73,41 @@ pub struct Output {
     pub unknown: BTreeMap<raw::Key, Vec<u8>>,
 }
 
+/// Taproot Tree representing a finalized [`TaprootBuilder`]
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct TapTree(pub(crate) TaprootBuilder);
+
+impl PartialEq for TapTree {
+    fn eq(&self, other: &Self) -> bool {
+        // Invariant: builder is finalized
+        self.node_info().hash.eq(&other.node_info().hash)
+    }
+}
+
+impl TapTree {
+
+    // get the inner node info as the builder is finalized
+    fn node_info(&self) -> &NodeInfo {
+        self.0.branch[0].as_ref().unwrap()
```

Comment:
`expect("why this is ok")`?

Reply:
This is builder algorithm invariant, finalized/complete builders have only 1 element in the branch field. We check this from_inner() method. Will add a comment and an expect